core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/CommandExtension.java

```diff
@@ -0,0 +1,46 @@
+package org.eclipse.dataspaceconnector.transfer.core;
+
+import org.eclipse.dataspaceconnector.core.CoreExtension;
+import org.eclipse.dataspaceconnector.core.ReflectiveObjectFactory;
+import org.eclipse.dataspaceconnector.core.system.InjectorImpl;
+import org.eclipse.dataspaceconnector.spi.ObjectFactory;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.system.InjectionPointScanner;
+import org.eclipse.dataspaceconnector.spi.system.Provides;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.transfer.command.CommandHandlerRegistry;
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+import org.eclipse.dataspaceconnector.transfer.core.commandqueue.CommandHandlerRegistryImpl;
+import org.eclipse.dataspaceconnector.transfer.core.commandqueue.handlers.CancelTransferCommandHandler;
+import org.eclipse.dataspaceconnector.transfer.core.commandqueue.handlers.DeprovisionCommandHandler;
+
+import static org.eclipse.dataspaceconnector.common.types.Cast.cast;
+
+/**
+ * Registers command handlers that the core provides
+ */
+@CoreExtension
+@Provides({ CommandHandlerRegistry.class })
+public class CommandExtension implements ServiceExtension {
+
+    @Inject
+    private TransferProcessStore store;
+    private ObjectFactory factory;
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        factory = new ReflectiveObjectFactory(new InjectorImpl(), new InjectionPointScanner(), context);
+
+        CommandHandlerRegistryImpl registry = new CommandHandlerRegistryImpl();
+        context.registerService(CommandHandlerRegistry.class, registry);
+
+        registerDefaultCommands(registry);
+    }
+
+    private void registerDefaultCommands(CommandHandlerRegistryImpl registry) {
+        registry.register(factory.constructInstance(cast(CancelTransferCommandHandler.class)));
```

Comment:
do we need all the `ObjectFactory` effort since we can simply register our command handlers implementation like the one at row 43? Less annotations and no unused constructors needed.

Reply:
we do not need it, it kind of "fell out" of the development as a by-product, so I wanted to show how it's used. Maybe a sample would be a better place for it though.